### PR TITLE
Jolt: Update to commit f094082aa, adding RISC-V, PPC64 and LoongArch support

### LINKED ISSUE #100557: Godot (Jolt) cannot be cross-compiled on Linux for ppc64 and rv64
### Tested versions

- Master branch as of (44dfa7e710572836cc9b2d88a90b2bb500a2502e)

### System information

Godot v4.4.dev (9ee41ad73) - Windows 10.0.26100 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated Radeon RX 580 Series (Advanced Micro Devices, Inc.; 31.0.21921.1000) - Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz (6 threads)

### Issue description

Since the merger of Jolt Physics (#99895), now Godot cannot be compiled nor cross-compiled for other architectures (PPC and RISC-V/RV64). It can only be compiled/cross-compiled for x86 and/or ARM.

For both, _it does_ say that the architecture is unsupported:
```bash
In file included from thirdparty/jolt_physics/Jolt/RegisterTypes.cpp:5:
In file included from thirdparty/jolt_physics/Jolt/Jolt.h:8:
thirdparty/jolt_physics/Jolt/Core/Core.h:206:3: error: Unsupported CPU architecture
  206 |         #error Unsupported CPU architecture
      |          ^
BUILD: Elapsed time [00:02:34]
Compiling thirdparty/jolt_physics/Jolt/AABBTree/AABBTreeBuilder.cpp ...
In file included from thirdparty/jolt_physics/Jolt/AABBTree/AABBTreeBuilder.cpp:5:
In file included from thirdparty/jolt_physics/Jolt/Jolt.h:8:
thirdparty/jolt_physics/Jolt/Core/Core.h:206:3: error: Unsupported CPU architecture
  206 |         #error Unsupported CPU architecture
      |          ^
thirdparty/jolt_physics/Jolt/Core/Core.h:458:34: error: use of undeclared identifier 'JPH_CPU_ADDRESS_BITS'
  458 | static_assert(sizeof(void *) == (JPH_CPU_ADDRESS_BITS == 64? 8 : 4), "Invalid size of pointer" );
      |                                  ^
In file included from thirdparty/jolt_physics/Jolt/RegisterTypes.cpp:5:
In file included from thirdparty/jolt_physics/Jolt/Jolt.h:12:
In file included from thirdparty/jolt_physics/Jolt/Core/Array.h:7:
thirdparty/jolt_physics/Jolt/Core/STLAllocator.h:47:63: error: use of undeclared identifier 'JPH_CPU_ADDRESS_BITS'
   47 |         static constexpr bool   needs_aligned_allocate = alignof(T) > (JPH_CPU_ADDRESS_BITS == 32? 8 : 16);
      |                                                                        ^
# ......
# .....
```

But this is kind of a bummer, more so for rv64 users.

### Steps to reproduce

1. Follow the [instructions for setting up an rv64 cross-compiler](https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_for_linuxbsd.html#cross-compiling-for-risc-v-devices).
2. Execute the build command there.
3. Wait for failure.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by akien-mga**
Fixes #100557.

### CODE REVIEW

**Review by mihe — APPROVED** [review]

File: modules/jolt_physics/config.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
 def can_build(env, platform):
-    return not env["disable_3d"]
+    if env["disable_3d"]:
+        return False
+
+    if env["arch"] in ["loongarch64", "ppc32", "ppc64", "rv64"]:
+        return False
+
+    return True
```
Comment (aaronfranke) [inline modules/jolt_physics/config.py:8]: Alternatively, we could enable it on supported architectures.

```suggestion
    return env["arch"] in ["x86_32", "x86_64", "arm32", "arm64", "wasm32"]:
```

**Comment by hpvb** [comment]
Jolt supports rv32 and rv64 as of https://github.com/jrouwe/JoltPhysics/pull/1400 we should note that and when we update Jolt update this list of archs. It'd be nice to support these as our friendly neighborhood open source cpu arch :)

**Comment by akien-mga** [comment]
I backported JoltPhysics#1400 to add RISC-V support, so now this PR only disables the module for `ppc32`/`ppc64`/`loongarch64`.

**Comment by akien-mga** [comment]
> > I backported JoltPhysics#1400 to add RISC-V support
> 
> Any reason why we wouldn't just update Jolt as a whole instead? The currently pinned version is largely just an arbitrary commit from Jolt's `master` branch, so isn't tied to any point release or anything like that.

Yeah we can do that too, I just haven't evaluated whether there's been any other upstream changes that might be a bit risky to use in production. Usually we prefer to stick to tagged releases but I see we're already on a dev commit so we might as well update further.

**Comment by mihe** [comment]
Might be good to bump it by one more commit, to jrouwe/JoltPhysics@27559da, just in case anyone runs into problems with the MSVC warning mentioned in jrouwe/JoltPhysics#1402.

**Comment by akien-mga** [comment]
Yeah, and I see https://github.com/jrouwe/JoltPhysics/pull/1377 was reopened so I was thinking of waiting for that one to be merged too to sync.

**Comment by jrouwe** [comment]
Yes, please give me a couple of days. I'll get PPC and LoongArch support in there too. I'll ping this thread when it's done.

**Comment by akien-mga** [comment]
Sounds good! No rush from our side, this is still a mostly theoretical ask and nothing urgent to solve.

**Comment by jrouwe** [comment]
PowerPC support has been merged, only Little Endian mode works atm. Big Endian compiles but fails the unit tests.

**Comment by jrouwe** [comment]
LoongArch support has also been added to Jolt. 

I think this PR can now be updated to the latest version of Jolt and merged.

**Comment by akien-mga** [comment]
Amazing work, thanks a lot @jrouwe!

The PR was updated to simply update Jolt Physics to https://github.com/jrouwe/JoltPhysics/commit/f094082aa2bbfcbebc725dbe8b8f65c7d5152886.

**Review by mihe — APPROVED** [review]
Looks good to me!

**Comment by jrouwe** [comment]
Was wondering why ppc32 was excluded, but I see now that it doesn't support 8 byte atomics, so this is indeed the best we can do.

**Review by aaronfranke — APPROVED** [review]
Looks good! By the way, in case anyone is wondering about `rv32`, the reason Godot has zero support for it is because this architecture is only used for microcontrollers. Since RISC-V is so new, there are no old 32-bit devices, every RISC-V device remotely capable of running Godot is 64-bit. There is also rv128, but supporting that would be a very very long journey (and Linux doesn't support it yet).

**Pull request merged**